pkg/azuredisk/controllerserver.go

```diff
@@ -524,7 +524,65 @@ func (d *Driver) GetCapacity(ctx context.Context, req *csi.GetCapacityRequest) (
 
 // ListVolumes return all available volumes
 func (d *Driver) ListVolumes(ctx context.Context, req *csi.ListVolumesRequest) (*csi.ListVolumesResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "")
+	var err error
+	start := 0
+	if req.StartingToken != "" {
+		start, err = strconv.Atoi(req.StartingToken)
+		if err != nil {
+			return nil, status.Errorf(codes.Aborted, "ListVolumes starting token(%s) parsing with error: %v", req.StartingToken, err)
+		}
+		if start < 0 {
+			return nil, status.Errorf(codes.Aborted, "ListVolumes starting token(%d) can not be negative", start)
+		}
+	}
+
+	disks, derr := d.cloud.DisksClient.ListByResourceGroup(ctx, d.cloud.ResourceGroup)
```

Comment:
actually there are other scenarios that disks are in created other resource group, so ListVolumes result is only accurate for default condition. On the other hand, agent nodes are all in `d.cloud.ResourceGroup`, so ListVolumes of all agent nodes would be accurate. I think this could be right solution: list disks of all agent nodes under `d.cloud.ResourceGroup`

Reply:
I am thinking use `d.cloud.KubeClient.CoreV1().PersistentVolumes()` to get all PV info and then get all diskURI from pv.csi.volumeid and `disk.ManagedBy`, that would provide complete disk list, could refer to:
 - `d.cloud.KubeClient.CoreV1()` interface
https://github.com/kubernetes-sigs/azurefile-csi-driver/blob/bb2550070c810ffe94c35eb38aad09c6a3136ade/pkg/azurefile/azurefile.go#L556

Reply:
this would require much more work, while it could list all disks this cluster is using, otherwise it could cover all disks and may provide incomplete disk list, not sure what would happen if disk list is incomplete.